superset-frontend/src/dashboard/components/DashboardBuilder/state.ts

```diff
@@ -75,11 +77,12 @@ export const useNativeFilters = () => {
     if (
       filterValues.length === 0 &&
       dashboardFiltersOpen &&
-      nativeFiltersEnabled
+      nativeFiltersEnabled &&
+      ['CONVERTED', 'REVIEWING', 'NOOP'].includes(filterboxMigrationState)
```

Comment:
this line fixed the dashboard left margin issue.
this is very tricky bug. I can reproduce this issue only when:
0. enable feature flag,
1. open dashboard list page,
2. click a dashboard.

if i open dashboard from its own dashboard link, can't see this left-margin problem :joy: